### PR TITLE
feat: Subprocess support for Ollma and vllm

### PR DESCRIPTION
**Pull request by Wendong-Fan**
## Description

1. launch vLLM and Ollama servers in a subprocess when no url is given

2. release: v 0.1.7.0

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/models/ollama_model.py [hunk 1]
```diff
@@ -48,17 +49,38 @@ def __init__(
...
+        try:
+            subprocess.Popen(
+                ["ollama", "server", "--port", "11434"],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+            print(
+                f"Ollama server started on http://localhost:11434/v1 "
+                f"for {self.model_type} model"
+            )
```
Comment (lightaime) [inline camel/models/ollama_model.py:80]: Missing a period.
```
            print(
                f"Ollama server started on http://localhost:11434/v1 "
                f"for {self.model_type} model."
            )
```

File: camel/models/ollama_model.py [hunk 2]
```diff
@@ -48,17 +49,38 @@ def __init__(
...
+        except Exception as e:
+            print(f"Failed to start Ollama server: {e}")
```
Comment (lightaime) [inline camel/models/ollama_model.py:82]: Missing a period.

File: camel/models/vllm_model.py [hunk 3]
```diff
@@ -52,17 +53,38 @@ def __init__(
...
+        try:
+            subprocess.Popen(
+                ["vllm", "server", "--port", "8000"],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+            print(
+                f"vllm server started on http://localhost:8000/v1 "
+                f"for {self.model_type} model"
+            )
```
Comment (lightaime) [inline camel/models/vllm_model.py:84]: Missing a period.

File: camel/models/vllm_model.py [hunk 4]
```diff
@@ -52,17 +53,38 @@ def __init__(
...
+        except Exception as e:
+            print(f"Failed to start vllm server: {e}")
```
Comment (lightaime) [inline camel/models/vllm_model.py:86]: Missing a period.

**Comment by Wendong-Fan** [comment]
thanks @lightaime , updated in https://github.com/camel-ai/camel/pull/930

**Pull request merged**